ios/RN/RNCamera.m

```diff
@@ -464,19 +452,45 @@ - (void)record:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)resolve r
       }
     }
 
-    dispatch_async(self.sessionQueue, ^{
-        [self updateFlashMode];
-        NSString *path = [RNFileSystem generatePathInDirectory:[[RNFileSystem cacheDirectoryPath] stringByAppendingPathComponent:@"Camera"] withExtension:@".mov"];
-        NSURL *outputURL = [[NSURL alloc] initFileURLWithPath:path];
-        [self.movieFileOutput startRecordingToOutputFileURL:outputURL recordingDelegate:self];
-        self.videoRecordedResolve = resolve;
-        self.videoRecordedReject = reject;
+    [self updateSessionAudioIsMuted:!!options[@"mute"]];
+
+    // @TODO: maxFileSize
+    self.canAppendBuffer = YES;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(self.maxDuration * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        self.canAppendBuffer = NO;
+
+        [self.writerInput markAsFinished];
+        [self.videoWriter finishWritingWithCompletionHandler:^{
+            if (self.videoWriter.status == AVAssetWriterStatusFailed) {
+                return reject(@"E_RECORDING_FAILED", @"An error occurred while recording a video.", nil);
+            }
+
+            resolve(@{ @"uri": self.videoWriter.outputURL.absoluteString });
+        }];
     });
 }
 
 - (void)stopRecording
 {
-    [self.movieFileOutput stopRecording];
+    self.canAppendBuffer = NO;
```

Comment:
As of today, when stopRecording is called, it resolves the `recordAsync` method promise. Right now, it looks like this does not happen here, so people with `const data = await this.camera.recordAsync()` would make it hang forever.

Reply:
@jgfidelis Correct, this is one of the areas that we need to bring back to parity with the existing behavior. Just didn't want to spend the time getting this perfect if the overall move to VideoDataOutput is not wanted :)